swift/Samples/Tutorial/Tutorial3.md

```diff
@@ -0,0 +1,645 @@
+# Step 3
+
+_State throughout a tree of workflows_
+
+## Setup
+
+Start from the completed code from Tutorial 2, or if you're jumping ahead, the completed `Tutorial2` pod.
+
+## Editing TODO items
+
+Now that a user can "log in" to their TODO list, we want to add the ability to edit the TODO items listed.
+
+### State ownership
+
+In the workflow framework, state flows _down_ the tree, and actions come _up_ as output events (as in the traditional computer science trees that grow downward).
+
+What this means is that state should be created as far down the tree as possible, to limit the scope of state to be as small as possible. Additionally, there should be only one "owner" of the state in the tree - if it's passed farther down the tree, it should be a copy or read-only version of it - so there is no shared mutable state in multiple workflows.
+
+When a child workflow has a copy of the state from its parent, it should change it by emitting an _output_ back to the parent, requesting that it be changed. The child will then receive an updated snapshot of the data from the parent - keeping ownership at a single level of the tree.
+
+This is all a bit abstract, so let's make it more concrete by adding an edit todo item workflow.
```

Comment:
Use the _stateless workflow_ terminology here? Was that part of the general techniques list that @zach-klippenstein put together, or is it just a Kotlin concept?

Reply:
It's more relevant in Kotlin, where there is a separate type for it